crates/red_knot_python_semantic/src/types/call/bind.rs

```diff
@@ -1,174 +1,199 @@
-                let Some((index, parameter)) = parameters
-                    .keyword_by_name(name)
-                    .or_else(|| parameters.keyword_variadic())
-                else {
-                    errors.push(CallBindingError::UnknownArgument {
-                        argument_name: ast::name::Name::new(name),
-                        argument_index: get_argument_index(argument_index, num_synthetic_args),
-                    });
-                    continue;
-                };
-                (index, parameter, ty, false)
-            }
+#[derive(Debug, Clone, PartialEq, Eq)]
+enum BindingsInner<'db> {
+    /// The call resolves to exactly one binding.
+    Single(CallableBinding<'db>),
 
-            Argument::Variadic(_) | Argument::Keywords(_) => {
-                // TODO
-                continue;
-            }
-        };
-        if let Some(expected_ty) = parameter.annotated_type() {
-            if !argument_ty.is_assignable_to(db, expected_ty) {
-                errors.push(CallBindingError::InvalidArgumentType {
-                    parameter: ParameterContext::new(parameter, index, positional),
-                    argument_index: get_argument_index(argument_index, num_synthetic_args),
-                    expected_ty,
-                    provided_ty: *argument_ty,
-                });
-            }
+    /// The call resolves to multiple bindings.
+    Union(Box<[CallableBinding<'db>]>),
+}
+
+impl<'db> Bindings<'db> {
+    /// Binds the arguments of a call site against a signature.
+    ///
+    /// The returned bindings provide the return type of the call, the bound types for all
+    /// parameters, and any errors resulting from binding the call, all for each union element and
+    /// overload (if any).
+    pub(crate) fn bind(
+        db: &'db dyn Db,
+        signatures: &'db Signatures<'db>,
+        arguments: &CallArguments<'_, 'db>,
+    ) -> Self {
+        if let Some(signature) = signatures.as_single() {
+            return Bindings {
+                signatures,
+                inner: BindingsInner::Single(CallableBinding::bind(db, signature, arguments)),
+            };
         }
-        if let Some(existing) = parameter_tys[index].replace(*argument_ty) {
-            if parameter.is_variadic() || parameter.is_keyword_variadic() {
-                let union = UnionType::from_elements(db, [existing, *argument_ty]);
-                parameter_tys[index].replace(union);
-            } else {
-                errors.push(CallBindingError::ParameterAlreadyAssigned {
-                    argument_index: get_argument_index(argument_index, num_synthetic_args),
-                    parameter: ParameterContext::new(parameter, index, positional),
-                });
-            }
+
+        let bindings = signatures
+            .iter()
+            .map(|signature| CallableBinding::bind(db, signature, arguments))
+            .collect::<Vec<_>>()
+            .into_boxed_slice();
```

Comment:
you can `.collect()` directly into a boxed slice:

```suggestion
            .collect();
```

Reply:
Done